xformers/components/attention/attention_mask.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+from typing import Optional, Type, TypeVar
+
+import torch
+
+Self = TypeVar("Self", bound="AttentionMask")
+
+
+class AttentionMask:
```

Comment:
tentatively own all attention masks, but this "breaks" (not per say, but not handled here) right now when things are sparsified